openverse_catalog/dags/maintenance/add_license_url.py

```diff
@@ -0,0 +1,179 @@
+    """Add license_url to meta_data batching all records with the same license.
+    :param postgres_conn_id: Postgres connection id
+    """
+
+    logger.info("Getting image records without license_url.")
+    postgres = PostgresHook(postgres_conn_id=postgres_conn_id)
+
+    total_count = 0
+    total_counts = {}
+    for license_items, path in license_map.items():
+        license_name, license_version = license_items
+        logger.info(f"Processing {license_name} {license_version}, {license_items}.")
+        license_url = f"{base_url}{path}/"
+
+        select_query = dedent(
+            f"""
+            SELECT identifier FROM image
+            WHERE (
+            meta_data is NULL AND license = '{license_name}'
+            AND license_version = '{license_version}')
+            """
+        )
+        result = postgres.get_records(select_query)
+
+        if not result:
+            logger.info(f"No records to update with {license_url}.")
+            continue
+        logger.info(f"{len(result)} records to update with {license_url}.")
+        license_url_col = {"license_url": license_url}
+        update_license_url_query = dedent(
+            f"""
+            UPDATE image
+            SET meta_data = {Json(license_url_col)}
+            WHERE identifier IN ({','.join([f"'{r[0]}'" for r in result])});
+            """
+        )
+
+        updated_count = postgres.run(
+            update_license_url_query, autocommit=True, handler=RETURN_ROW_COUNT
+        )
+        logger.info(f"{updated_count} records updated with {license_url}.")
+        total_counts[license_url] = updated_count
+        total_count += updated_count
+
+    logger.info(f"{total_count} image records with missing license_url updated.")
+    logger.info(f"Total updated counts: {total_counts}")
+    return total_counts
+
+
+def final_report(postgres_conn_id: str, item_count):
+    logger.info(
+        "Added license_url to all items. Checking for any records "
+        "that still have NULL metadata."
+    )
+    postgres = PostgresHook(postgres_conn_id=postgres_conn_id)
+    null_meta_data_records = postgres.get_first(
+        dedent("SELECT COUNT(*) from image WHERE meta_data IS NULL;")
+    )[0]
+    logger.info(f"There are {null_meta_data_records} records with NULL meta_data.")
+
+    message = f"""
+Added license_url to *{item_count}* items`
+Now, there are {null_meta_data_records} records with NULL meta_data.
+"""
+    send_message(
+        message,
+        username="Airflow DAG Data Normalization - license_url",
+        dag_id="data_normalization",
+    )
+
+    logger.info(message)
+
+
+DAG_DEFAULT_ARGS = {
+    "owner": "data-eng-admin",
+    "depends_on_past": False,
+    "start_date": datetime(2020, 6, 15),
+    "template_undefined": jinja2.Undefined,
+    "email_on_retry": False,
+    "schedule_interval": None,
+    "retries": 0,
+    "on_failure_callback": slack.on_failure_callback,
+}
```

Comment:
Should this be here? I think we can inherit from here and extend where necessary:
https://github.com/WordPress/openverse-catalog/blob/895e907b6f9594b4d43db6a07eae0b3cb4a0e7d0/openverse_catalog/dags/common/constants.py#L16-L29

Reply:
I've used the constant, and have a question about the "schedule_interval": should it be in the `default_args` or the DAG args? Currently, I have both, don't know why.

Reply:
The `default_args` are what get passed to each task within the DAG, so the schedule interval should be provided to the `Dag` itself directly!